### PR TITLE
allow setting method and status of a PUSH

### PR DESCRIPTION
**Pull request by bmeck**

### CODE REVIEW

**Comment by jcrugzz** [comment]
This makes sense to me 👍

File: lib/spdy/response.js [hunk 1]
```diff
@@ -78,8 +78,8 @@ exports.end = function end(data, encoding, callback) {
 exports.push = function push(path, headers, callback) {
   var frame = {
     path: path,
-    method: 'GET',
-    status: 200,
+    method: typeof headers.method === 'string' ? headers.method : 'GET',
```
Comment (indutny) [inline lib/spdy/response.js]: Do we really want to be that restrictive? Perhaps, just `headers.method ? headers.method.toString() : 'GET'`

Comment (bmeck) [inline lib/spdy/response.js]: sounds fine, was just trying to match the typeof w/ `reason` earlier in the file

Comment (indutny) [inline lib/spdy/response.js]: Oh, it is there for a different reason (and that code is a copy-paste from core). Let's remove `typeof`. Thanks!

**Comment by indutny** [comment]
Landed and released, thank you!

**Pull request closed without merging**